### DISCUSSION TITLE
How to use bookmarks on Android?

### DISCUSSION THREAD

**Discussion by pmorch** [nextcloud/bookmarks#2055] [Q&A]
Hi,

I have not been able to find an app or approach that allows me to add bookmarks on Android with these requriements:

* Must be an Android share target.
* Must be able to automatically get the title from the source page.
* Must work on a Phone.
* Must be able to handle 4K+ bookmarks.
* Must not require adding/trusting an app store in addition to Google Play.

This is a shame, because I add 90% of my bookmarks on my phone, and so this makes Nextcould bookmarks unusable for me. Did I miss a way to do it that could work? Here are the issues I ran into:

* Raw HTML interface: #2030 - I could not view or edit the tags. I use tags extensively.
* [Nextcloud Bookmarks](https://gitlab.com/bisada/OCBookmarks): [[Feature Request] Make Nextcloud Bookmarks available on Google Play store (#105)](https://gitlab.com/bisada/OCBookmarks/-/issues/105) I don't want to trust F-droid or Amazon's store, even if I could get them to work, which I didn't try.
* [NCBookmark](https://gitlab.com/lenchan139/NCBookmark) - This says it is a viewer only. Last commit 4 years ago.
* [Google Play Store - Bookmarks for Nextcloud](https://play.google.com/store/apps/details?id=de.emasty.bookmarks)
  * When I add a bookmark, it doesn't get the title of the URL, only the URL itself. I want the title also.
  * The Google Play page does not list any method to file bugs or any links to github or gitlab.
  *  https://emasty.de does not have a valid HTTPS certificate.
  * So I added a review instead.
* [Floccus](https://floccus.org): I hit what apparently is a known issue with many bookmarks (I have 4k+) so I added the details of my experience in https://github.com/floccusaddon/floccus/issues/1244#issuecomment-1667720854. Floccus was completely unusable for me.

So I think I've tried all the listed options and it doesn't look like there is any option that allows me to save bookmarks from Android with title and URL unless I'm willing to trust the F-droid or Amazon stores, and I'm not.

Is there anything I have missed?

Otherwise I'll try some another self-hosted bookmarks solution. Which is a shame since I already have nextcloud installed.

Is anybody able to recommend a good alternative?

**Comment by marcelklehr** [comment]
https://github.com/nextcloud/bookmarks/issues/2030 is a relatively minor bug, I expect I will get around to fixing it soon :)
As for floccus, the memory problems with larger collections are less easily fixed, although I also have a plan for that :)

**Reply by marcelklehr** [reply]
...aaand v13.1.0 is out with lots of fixes, including #2030 :)

**Comment by emasty** [comment]
Hello @pmorch,

I am the author of the Android app "Bookmarks for Nextcloud".
The app should check all your requirements. Also 4k+ bookmarks shouldn't be an issue.

Adding the title automatically has to be enabled in the backend settings. Once the setting is enabled, the title should be filled automatically.

The setting is available in the admin settings at Administration -> Bookmarks -> Privacy

Please let me know if that solves your issue or if you have any further feedback or questions!

Best regards, Marcus
